ibis/backends/duckdb/__init__.py

```diff
@@ -1180,7 +1180,7 @@ def _get_temp_view_definition(
     def _register_udfs(self, expr: ir.Expr) -> None:
         import ibis.expr.operations as ops
 
-        with self.begin() as con:
+        with self.con.connect() as con:
```

Comment:
For some reason in duckdb 0.9, this `begin` ends up creating a nested transaction, I'm not entirely sure why (or if that's manifestation of some other issue either in ibis or the relevant dependencies)